src/cache/gcs.rs

```diff
@@ -229,14 +229,14 @@ impl GCSCredentialProvider {
             issued_at: chrono::UTC::now().timestamp(),
         };
 
-        let binary_key = openssl::rsa::Rsa::private_key_from_pem(
+        let pem_cert = pem::parse(
             self.sa_key.private_key.as_bytes()
-        )?.private_key_to_der()?;
+        )?;
 
         let auth_request_jwt = jwt::encode(
             &jwt::Header::new(jwt::Algorithm::RS256),
             &jwt_claims,
-            &binary_key,
+            &pem_cert.contents,
```

Comment:
It looks to me like the private key is never converted to DER format, like it was above.

Reply:
Or is that just the same as decoding the base64 to binary?

Reply:
@fxb that's right (assuming you stripped the `--- BEGIN... ` `---END..` pre/post ambles first), there is a good overview at https://knowledge.digicert.com/generalinformation/INFO4448.html about the differences